google/cloud/bigquery/table.py

```diff
@@ -73,7 +74,8 @@
     # Unconditionally import optional dependencies again to tell pytype that
     # they are not None, avoiding false "no attribute" errors.
     import pandas
-    import pyarrow
+
+    pyarrow = _pyarrow_helpers.PYARROW_VERSIONS.try_import()
```

Comment:
This doesn't need to be `try_import` as we run our type checker only in an environment where pyarrow is installed.

Likely we'd just hit `None` has no attribute XYZ if we end up in this code path.

Reply:
reverted